### PR TITLE
Image writer

### PR DESCRIPTION
**Pull request by elevin72**
Pllz looook

### CODE REVIEW

**Review by abrahammurciano — CHANGES_REQUESTED** [review]
LMK when you want me to review it again when u push the stuff u changed when we were on call before.

File: src/main/rendering/ImageWriter.java [hunk 1]
```diff
@@ -1,21 +1,39 @@
...
- * @author
- * @author
  */
 public class ImageWriter {
+
+	/**  
+	 * The path where the image will be stored
+	 */
+	public static final String PATH = "images/";
+	public static final String EXT = ".png";
```
Comment (abrahammurciano) [inline src/main/rendering/ImageWriter.java:20]: Shouldn't we take this stuff as arguments? That way the user could specify any output in any folder with any file extension?

The argument `filename` should really be called `filepath` and should contain all this stuff.

File: src/main/rendering/ImageWriter.java [hunk 2]
```diff
@@ -26,13 +44,18 @@ public ImageWriter(String filename, Resolution resolution) {
...
+		pixels[row * width + col] = colour.rgb();
 	}
 
 	/**
 	 * Write the image to disk.
 	 */
 	public void writeToFile() {
-		// TODO: implement
+		try {
+			ImageIO.write(image, "PNG", new File( PATH + filename + EXT));
```
Comment (abrahammurciano) [inline src/main/rendering/ImageWriter.java:55]:
```suggestion
			ImageIO.write(image, "JPG", new File(filename));
```
As per my previous comment.
Also PNG supports transparent background, thus JPG is more efficient if we don't need that. Any reason why you changed to PNG?

File: src/test/unit/rendering/ImageWriterTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+				
+			}
+		}
+		iw.writeToFile();
+		assertImageCorrect("Error Message", fileName);
+	}
+
+	private void assertImageCorrect(String errorMessage, String fileName) {
+		JFrame frame = new JFrame();
+		JLabel label = new JLabel(new ImageIcon(ImageWriter.PATH + fileName + ImageWriter.EXT));
```
Comment (abrahammurciano) [inline src/test/unit/rendering/ImageWriterTest.java:40]:
```suggestion
		JLabel label = new JLabel(new ImageIcon(fileName));
```
Again, filename could contain the whole path, no need to hard code the output directory in the imagewriter

**Review by abrahammurciano — COMMENTED** [review]
Good work!

File: src/main/rendering/ImageWriter.java (hunk 1 above)
Comment (elevin72) [reply src/main/rendering/ImageWriter.java:20]: Could be. Instructions said to put it in images/
But that could be for just this exercise.

File: src/main/rendering/ImageWriter.java (hunk 2 above)
Comment (elevin72) [reply src/main/rendering/ImageWriter.java:55]: Again, instructions. Doesn't really matter to me.

File: src/main/rendering/ImageWriter.java (hunk 1 above)
Comment (abrahammurciano) [reply src/main/rendering/ImageWriter.java:20]: Since when do we follow orders?

**Review by abrahammurciano — APPROVED** [review]
Great work!

**Comment by abrahammurciano** [comment]
Close #24

**Pull request merged**